SwiftUiMovieDb/Modules/Home/Views/MovieListView.swift

```diff
@@ -9,42 +9,44 @@ import SwiftUI
 
 struct MovieListView: View {
     
-    @ObservedObject private var movieListState = MovieListViewModel()
+    @StateObject private var manager = MovieListViewModel()
     
     var body: some View {
         NavigationView {
             List {
-                movieSection(title: "Now Playing", movieType: .now_playing)
-                movieSection(title: "Upcoming", movieType: .upcoming)
-                movieSection(title: "Top Rated", movieType: .top_rated)
-                movieSection(title: "Popular", movieType: .popular)
+                ForEach(MovieListType.allCases, id: \.self) {
+                    movieSection(movieType: $0)
+                }
             }
             .navigationBarTitle("The MovieDb")
         }
         .onAppear {
-            self.movieListState.loadMovies(with: .now_playing)
-            self.movieListState.loadMovies(with: .upcoming)
-            self.movieListState.loadMovies(with: .top_rated)
-            self.movieListState.loadMovies(with: .popular)
+            self.manager.loadMovies(with: .now_playing)
+            self.manager.loadMovies(with: .upcoming)
+            self.manager.loadMovies(with: .top_rated)
+            self.manager.loadMovies(with: .popular)
         }
     }
-
-private func movieSection(title: String, movieType: MovieListType) -> some View {
-    Group {
-        if movieListState.movies != nil {
-            if movieType == .now_playing {
-                MoviePosterCarouselView(title: title, movies: movieListState.movies!)
+    
+    private func movieSection(movieType: MovieListType) -> some View {
+        Group {
+            if manager.movies.isEmpty {
```

Comment:
why to show loader when movies array is empty ? rather show some view saying no content available.